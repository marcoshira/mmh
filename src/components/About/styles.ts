import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding-top: ${theme.spacings.medium};
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    ${Title} {
      /* margin-top: ${theme.spacings.ltmedium}; */
    }
  `}
`;

export const ContentWrapper = styled.div`
  display: flex;
  flex-flow: row nowrap;
  width: 80%;
  img {
    width: 620px;
    height: 500px;
  }
`;

export const TextWrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.huge};
    padding-left: ${theme.spacings.huge};
    /* padding-top: ${theme.spacings.huge}; */
    margin-right: 0;
    line-height: 1.5;
    font-size: ${theme.font.sizes.medium};
  `}
`;
