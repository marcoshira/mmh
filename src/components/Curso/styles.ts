import styled, { css } from 'styled-components';

export const Wrapper = styled.span`
  ${({ theme }) => css`
    width: 60%;
    height: 38px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    padding: ${theme.spacings.xsmall} 0;
    border: 1.5px solid ${theme.colors.secondaryColor};
    border-radius: 10px;
    position: relative;
    a {
      text-decoration: none;
      position: absolute;
      right: 20px;
      &:hover {
        color: ${theme.colors.secondaryColor};
      }
    }

    span {
      position: absolute;
      left: 20px;
    }
  `}
`;
