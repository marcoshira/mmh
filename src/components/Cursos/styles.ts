import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    padding-top: ${theme.spacings.medium};
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    justify-content: center;
  `}
`;

export const CursosWrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
