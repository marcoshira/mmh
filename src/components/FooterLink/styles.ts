import styled, { css } from 'styled-components';

type showHover = {
  showHover?: boolean;
};

const hover = (theme) => css`
  color: ${theme.colors.thirdColor};
`;

export const Wrapper = styled.div<showHover>`
  ${({ theme, showHover }) => css`
    font-size: ${theme.font.sizes.ltmedium};
    margin-left: ${theme.spacings.medium};
    img {
      width: ${theme.spacings.medium};
      height: ${theme.spacings.medium};
    }
    a {
      display: flex;
      text-decoration: none;
      color: ${theme.colors.primaryColor};
      height: ${theme.spacings.medium};

      &:hover {
        color: ${theme.colors.primaryColor};
        ${showHover && hover(theme)}
      }
    }

    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.medium};
      margin-left: 0;
    }
  `}
`;

export const LinkWrapper = styled.span`
  ${({ theme }) => css`
    position: relative;
    left: ${theme.spacings.xsmall};
    top: 0px;
    color: ${theme.colors.footerPrimaryColor};
  `}
`;

export const NoLinkWrapper = styled.a`
  ${({ theme }) => css`
    position: relative;
    left: ${theme.spacings.xsmall};
    top: 0px;

    @media ${theme.media.lteMedium} {
      left: 0px;
    }
  `}
`;
