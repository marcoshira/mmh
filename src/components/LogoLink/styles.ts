import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: ${theme.colors.thirdColor};

    &:hover {
      color: ${theme.colors.secondaryColor};
    }

    > img {
      height: 3rem;
    }
  `};
`;
