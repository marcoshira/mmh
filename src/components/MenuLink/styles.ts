import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.font.sizes.ltmedium};
    padding: ${theme.spacings.small};
    color: ${theme.colors.thirdColor};
    position: relative;
    transition: all 200ms ease-in-out;

    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondaryColor};
      transition: all 200ms ease-in-out;
    }

    &:hover::after {
      left: 25%;
      width: 50%;
    }

    &:hover {
      color: ${theme.colors.secondaryColor};
    }
  `}
`;
