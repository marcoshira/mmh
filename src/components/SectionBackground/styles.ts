import styled, { css } from 'styled-components';

type Background = {
  background?:
    | 'crimson'
    | 'blue'
    | 'navy'
    | 'white'
    | 'black'
    | 'pink'
    | 'purple'
    | 'orange'
    | 'green'
    | 'lightBlue'
    | 'yellow'
    | 'brown'
    | 'grey'
    | 'oi';
};

export const Container = styled.div<Background>`
  ${({ theme, background, color }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${backgroundcolor(theme, background)};
    ${textcolor(theme, color)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;

const backgroundcolor = (theme, background) => {
  if (background === 'crimson') {
    return css`
      background: ${theme.colors.crimson};
    `;
  }
  if (background === 'white') {
    return css`
      background: ${theme.colors.white};
    `;
  }
  if (background === 'blue') {
    return css`
      background: ${theme.colors.blue};
    `;
  }
  if (background === 'navy') {
    return css`
      background: ${theme.colors.navy};
    `;
  }
  if (background === 'black') {
    return css`
      background: ${theme.colors.black};
    `;
  }
  if (background === 'pink') {
    return css`
      background: ${theme.colors.pink};
    `;
  }
  if (background === 'purple') {
    return css`
      background: ${theme.colors.purple};
    `;
  }
  if (background === 'orange') {
    return css`
      background: ${theme.colors.orange};
    `;
  }
  if (background === 'green') {
    return css`
      background: ${theme.colors.green};
    `;
  }
  if (background === 'lightBlue') {
    return css`
      background: ${theme.colors.lightBlue};
    `;
  }
  if (background === 'yellow') {
    return css`
      background: ${theme.colors.yellow};
    `;
  }
  if (background === 'brown') {
    return css`
      background: ${theme.colors.brown};
    `;
  }
  if (background === 'grey') {
    return css`
      background: ${theme.colors.grey};
    `;
  } else {
    return css`
      background: ${theme.colors.white};
    `;
  }
};

export const textcolor = (theme, color) => {
  if (color === 'crimson') {
    return css`
      color: ${theme.colors.crimson};
    `;
  }
  if (color === 'white') {
    return css`
      color: ${theme.colors.white};
    `;
  }
  if (color === 'blue') {
    return css`
      color: ${theme.colors.blue};
    `;
  }
  if (color === 'navy') {
    return css`
      color: ${theme.colors.navy};
    `;
  }
  if (color === 'black') {
    return css`
      color: ${theme.colors.black};
    `;
  }
  if (color === 'pink') {
    return css`
      color: ${theme.colors.pink};
    `;
  }
  if (color === 'purple') {
    return css`
      color: ${theme.colors.purple};
    `;
  }
  if (color === 'orange') {
    return css`
      color: ${theme.colors.orange};
    `;
  }
  if (color === 'green') {
    return css`
      color: ${theme.colors.green};
    `;
  }
  if (color === 'lightBlue') {
    return css`
      color: ${theme.colors.lightBlue};
    `;
  }
  if (color === 'yellow') {
    return css`
      color: ${theme.colors.yellow};
    `;
  }
  if (color === 'brown') {
    return css`
      color: ${theme.colors.brown};
    `;
  }
  if (color === 'grey') {
    return css`
      color: ${theme.colors.grey};
    `;
  } else {
    return css`
      color: ${theme.colors.navy};
    `;
  }
};
