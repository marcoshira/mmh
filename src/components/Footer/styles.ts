import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.thirdColor};
    margin: ${theme.spacings.xlarge} 0;

    ${SectionContainer} {
      > span {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: space-evenly;

    margin-top: ${theme.spacings.large};
  `}
`;
