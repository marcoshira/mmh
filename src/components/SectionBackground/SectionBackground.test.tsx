import { SectionBackground } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<SectionBackground />', () => {
  it('should render without background and text color and match snapshot', () => {
    const { container } = renderTheme(
      <SectionBackground background={undefined} color={undefined}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with else background and text color and match snapshot', () => {
    const { container } = renderTheme(
      <SectionBackground background={'oi'} color={'oi'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with background and text color default and match snapshot', () => {
    const { container } = renderTheme(
      <SectionBackground>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.white,
      color: theme.colors.navy,
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with background and text color wrong thus default and match snapshot', () => {
    const { container } = renderTheme(
      <SectionBackground background={undefined} color={undefined}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.white,
      color: theme.colors.navy,
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with background and text color crimson and match snapshot', () => {
    const { container } = renderTheme(
      <SectionBackground background={'crimson'} color={'crimson'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.crimson,
      color: theme.colors.crimson,
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with background and text color blue', () => {
    const { container } = renderTheme(
      <SectionBackground background={'blue'} color={'blue'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.blue,
      color: theme.colors.blue,
    });
  });

  it('should render with background and text color white', () => {
    const { container } = renderTheme(
      <SectionBackground background={'white'} color={'white'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.white,
      color: theme.colors.white,
    });
  });

  it('should render with background and text color navy', () => {
    const { container } = renderTheme(
      <SectionBackground background={'navy'} color={'navy'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.navy,
      color: theme.colors.navy,
    });
  });

  it('should render with background and text color black', () => {
    const { container } = renderTheme(
      <SectionBackground background={'black'} color={'black'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.black,
      color: theme.colors.black,
    });
  });

  it('should render with background and text color orange', () => {
    const { container } = renderTheme(
      <SectionBackground background={'orange'} color={'orange'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.orange,
      color: theme.colors.orange,
    });
  });

  it('should render with background and text color green', () => {
    const { container } = renderTheme(
      <SectionBackground background={'green'} color={'green'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.green,
      color: theme.colors.green,
    });
  });

  it('should render with background and text color lightBlue', () => {
    const { container } = renderTheme(
      <SectionBackground background={'lightBlue'} color={'lightBlue'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.lightBlue,
      color: theme.colors.lightBlue,
    });
  });

  it('should render with background and text color purple', () => {
    const { container } = renderTheme(
      <SectionBackground background={'purple'} color={'purple'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.purple,
      color: theme.colors.purple,
    });
  });

  it('should render with background and text color pink', () => {
    const { container } = renderTheme(
      <SectionBackground background={'pink'} color={'pink'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.pink,
      color: theme.colors.pink,
    });
  });

  it('should render with background and text color yellow', () => {
    const { container } = renderTheme(
      <SectionBackground background={'yellow'} color={'yellow'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.yellow,
      color: theme.colors.yellow,
    });
  });

  it('should render with background and text color brown', () => {
    const { container } = renderTheme(
      <SectionBackground background={'brown'} color={'brown'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.brown,
      color: theme.colors.brown,
    });
  });

  it('should render with background and text color grey', () => {
    const { container } = renderTheme(
      <SectionBackground background={'grey'} color={'grey'}>
        <h1>Children</h1>
      </SectionBackground>,
    );
    expect(container.firstChild).toHaveStyle({
      background: theme.colors.grey,
      color: theme.colors.grey,
    });
  });
});
