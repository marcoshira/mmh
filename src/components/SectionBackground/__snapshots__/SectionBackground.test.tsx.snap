// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with background and text color crimson and match snapshot 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: crimson;
  color: crimson;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
    color="crimson"
    id=""
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with background and text color default and match snapshot 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
    color="navy"
    id=""
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with background and text color wrong thus default and match snapshot 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
    color="navy"
    id=""
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with else background and text color and match snapshot 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
    color="oi"
    id=""
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;

exports[`<SectionBackground /> should render without background and text color and match snapshot 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
    color="navy"
    id=""
  >
    <div
      class="c1"
    >
      <h1>
        Children
      </h1>
    </div>
  </div>
</div>
`;
